Add "How to make a libxkbcommon release" (#560)
xkbcommon/libxkbcommon: RELEASING.md

# How to make a libxkbcommon release

### Prerequisites

- Have write access to xkbcommon Git repositories.
- Be subscribed to the [wayland-devel](https://lists.freedesktop.org/mailman/listinfo/wayland-devel) mailing list.

### Steps

#### Prepare the release

> - Update the `NEWS.md` file for the release, following the pattern of previous releases.

Reviewer: ```suggestion
- Update the `NEWS.md` file for the release, following [the corresponding instructions](changes/README.md).
```

Author: Updated, thanks.

# How to make a libxkbcommon release

### Prerequisites

- Have write access to xkbcommon Git repositories.
- Be subscribed to the [wayland-devel](https://lists.freedesktop.org/mailman/listinfo/wayland-devel) mailing list.

### Steps

#### Prepare the release

> - [ ] Update the `NEWS.md` file for the release, following [the corresponding instructions](changes/README.md).

Reviewer: ```suggestion
- [ ] Create a release branch: `git checkout -b release/vMAJOR.MINOR.PATCH master`

- [ ] Update the `NEWS.md` file for the release, following [the corresponding instructions](changes/README.md).
```

- [ ] Bump the `version` in `meson.build`.

- [ ] Run `meson dist -C build` to make sure the release is good to go.

> - [ ] Commit `git commit -m 'Bump version to MAJOR.MINOR.PATCH'`.

Reviewer: ```suggestion
- [ ] Commit `git commit -m 'Bump version to MAJOR.MINOR.PATCH'`.

- [ ] Create a merge request using this template and ensure *all* CI is green.
```

> - [ ] Tag `git tag --anotated -m xkbcommon-<MAJOR.MINOR.PATCH> xkbcommon-<MAJOR.MINOR.PATCH>`.

Reviewer: ```suggestion
- [ ] Tag `git tag --annotate -m xkbcommon-<MAJOR.MINOR.PATCH> xkbcommon-<MAJOR.MINOR.PATCH>`.
```

- [ ] Push `git push --follow-tags`.

#### Send announcement email to wayland-devel

- [ ] Send an email to the wayland-devel@lists.freedesktop.org mailing list, using this template:

```
Subject: [ANNOUNCE] libxkbcommon MAJOR.MINOR.PATCH

<NEWS & comments for this release>

Git tag:
--------

git tag: xkbcommon-<MAJOR.MINOR.PATCH>
git commit: <git commit sha>

<YOUR NAME>
```

#### Update website

- [ ] Pull the latest [website repository](https://github.com/xkbcommon/website).

> - [ ] Add the doc for the release: `cp -r <xkbommon>/build/html doc/<MAJOR.MINOR.PATCH>`.

Reviewer: We should define a range of versions for Doxygen. It is not that reliable, things (e.g. links) got broken/repaired multiple times.

IIRC our runners use a version that is bit old and there are some issues in the display.

Reviewer: That would be the best way indeed. But the issue is that Doxygen breaks things regularly (they do not have much regression test suite), so we need some manual control.

E.g. currently I have the latest version (1.12.0) and I get these warnings:
```
...
home/user/Repositories/libxkbcommon/include/xkbcommon/xkbcommon.h:1775: warning: unable to resolve reference to 'virtual-modifier-def' for \ref command
/home/user/Repositories/libxkbcommon/include/xkbcommon/xkbcommon.h:1797: warning: unable to resolve reference to 'virtual-modifier-def' for \ref command
...
```

This is fixed in current master, which we could use if generating the doc manually.

All in all I would like to drop Doxygen, but it seems there is no replacement that’s as complete, easy to install and free.

Author: > E.g. currently I have the latest version (1.12.0) and I get these warnings:

Indeed I have seen those.

> All in all I would like to drop Doxygen, but it seems there is no replacement that’s as complete, easy to install and free.

In the mean time, let's try to publish the docs with a good build and hope they release a new version soon...

# How to make a libxkbcommon release

### Prerequisites

- Have write access to xkbcommon Git repositories.
- Be subscribed to the [wayland-devel](https://lists.freedesktop.org/mailman/listinfo/wayland-devel) mailing list.

### Steps

#### Prepare the release

> - [ ] Create a release branch: `git checkout -b release/vMAJOR.MINOR.PATCH master`

Author: Added

  must be *explicitly* postponed by changing their milestone.

- [ ] Create a release branch: `git checkout -b release/vMAJOR.MINOR.PATCH master`

- [ ] Update the `NEWS.md` file for the release, following [the corresponding instructions](changes/README.md).

- [ ] Bump the `version` in `meson.build`.

- [ ] Run `meson dist -C build` to make sure the release is good to go.

- [ ] Commit `git commit -m 'Bump version to MAJOR.MINOR.PATCH'`.

- [ ] Create a pull request using this template and ensure *all* CI is green.

- [ ] Merge the pull request.

- [ ] Tag `git pull && git tag --annotate -m xkbcommon-<MAJOR.MINOR.PATCH> xkbcommon-<MAJOR.MINOR.PATCH>`.

- [ ] Push the tag `git push xkbcommon-<MAJOR.MINOR.PATCH>`.

#### Send announcement email to wayland-devel

- [ ] Send an email to the wayland-devel@lists.freedesktop.org mailing list, using this template:

```
Subject: [ANNOUNCE] libxkbcommon MAJOR.MINOR.PATCH

<NEWS & comments for this release>

Git tag:
--------

git tag: xkbcommon-<MAJOR.MINOR.PATCH>
git commit: <git commit sha>

<YOUR NAME>
```

#### Update website

- [ ] Pull the latest [website repository](https://github.com/xkbcommon/website).

> - [ ] Add the doc for the release: `cp -r <xkbommon>/build/html doc/<MAJOR.MINOR.PATCH>`.

Reviewer: ```suggestion
- [ ] Add the doc for the release: `cp -r <xkbommon>/build/html doc/<MAJOR.MINOR.PATCH>`. Check carefully that there is no warning during generation with Doxygen. It may be necessary to use another version of Doxygen to get a clean build. Building from source using the main branch is also a good option.
```

From our discussion above. The main point it to warn to check the Doxygen generation, feel free to remove the rest.

Author: Added!
